### PR TITLE
Add generic Cradle interface

### PR DESCRIPTION
**Pull request by Coobaha**
It allows to refine Cradle interface when used:

```typescript
declare module 'fastify-awilix' {
  interface Cradle {
    authService: AuthService;
  }
}
//later, type is inferred correctly
fastify.diContainer.cradle.authService
```

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by kibertoad** [comment]
@Coobaha Thanks! Can you also link awilix documentation on Cradle (https://github.com/jeffijoe/awilix#containercradle) and maybe add one sentence summary for those too lazy to click, since it's not exactly a self-explanatory concept?

**Comment by Coobaha** [comment]
done 👍

File: index.test-d.ts [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 import fastify, { FastifyInstance } from "fastify";
-import { diContainer, FastifyAwilixOptions, fastifyAwilixPlugin } from './'
+import { diContainer, FastifyAwilixOptions, fastifyAwilixPlugin, Cradle } from './'
 import { expectAssignable, expectType } from 'tsd'
 
 expectAssignable<FastifyAwilixOptions>({})
 expectAssignable<FastifyAwilixOptions>({ disposeOnClose: false})
 expectAssignable<FastifyAwilixOptions>({ disposeOnResponse: false})
-expectType<AwilixContainer>(diContainer)
+expectType<AwilixContainer<Cradle>>(diContainer)
```
Comment (kibertoad) [inline index.test-d.ts:9]: Could you add interface extension in this file and actually assert that extension takes effect both for this and for requestcradle?

Comment (Coobaha) [reply index.test-d.ts:26]: Great suggestion 👌 , I've also used this file to provide simple usage example for users

**Comment by kibertoad** [comment]
Thanks a lot!

**Comment by kibertoad** [comment]
Released in 1.1.0

**Pull request merged**